tests/dogtag/pytest-ansible/pytest/performance_test/test_acme_cert_enrollment.py

```diff
@@ -0,0 +1,235 @@
+import hashlib
+import re, logging
+import threading
+import OpenSSL
+from acme import crypto_util
+from timeit import default_timer as timer
+from urllib.request import urlopen, Request
+
+DEFAULT_DIRECTORY_URL = "http://localhost:8080/acme/directory"
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.StreamHandler())
+LOGGER.setLevel(logging.INFO)
+
+parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter)
+parser.add_argument("--domain", required=True, help="Client Domain name")
+parser.add_argument("--acme-dir", required=True, help="path to the .well-known/acme-challenge/ directory")
+parser.add_argument("--directory-url", required=True, default=DEFAULT_DIRECTORY_URL,
+                    help="certificate authority directory url")
+parser.add_argument("--number-of-threads", required=True, help="Number of threads", type=int)
+parser.add_argument("--number-of-tests-per-thread", required=True, help="Number of test per thread", type=int)
+
+args = parser.parse_args()
+LOGGER.setLevel(LOGGER.level)
+
+account_key = "{}.key".format(args.domain)
+csr = "{}.csr".format(args.domain)
+
+
+def new_csr_comp(domain_name, pkey_pem=None):
+    """Create certificate signing request."""
+    if pkey_pem is None:
+        # Create private key.
+        pkey = OpenSSL.crypto.PKey()
+        pkey.generate_key(OpenSSL.crypto.TYPE_RSA, 2048)
+        pkey_pem = OpenSSL.crypto.dump_privatekey(OpenSSL.crypto.FILETYPE_PEM, pkey)
+        kf = open("{}.key".format(domain_name), "wb")
+        kf.write(pkey_pem)
+        kf.close()
+    csr_pem = crypto_util.make_csr(pkey_pem, [domain_name])
+    cf = open("{}.csr".format(domain_name), "wb")
+    cf.write(csr_pem)
+    cf.close()
+    return pkey_pem, csr_pem
+
+
+# helper functions - base64 encode for jose spec
+def b64(b):
+    return base64.urlsafe_b64encode(b).decode('utf8').replace("=", "")
+
+
+# helper function - run external commands
+def cmd(cmd_list, stdin=None, cmd_input=None, err_msg="Command Line Error"):
+    proc = subprocess.Popen(cmd_list, stdin=stdin, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    out, err = proc.communicate(cmd_input)
+    if proc.returncode != 0:
+        raise IOError("{0}\n{1}".format(err_msg, err))
+    return out
+
+
+# helper function - make request and automatically parse json response
+def do_request(url, data=None, err_msg="Error"):
+    try:
+        resp = urlopen(
+            Request(url, data=data, headers={"Content-Type": "application/jose+json", "User-Agent": "acme-cert"}))
+        resp_data, code, headers = resp.read().decode("utf8"), resp.getcode(), resp.headers
+    except IOError as e:
+        resp_data = e.read().decode("utf8") if hasattr(e, "read") else str(e)
+        code, headers = getattr(e, "code", None), {}
+    try:
+        resp_data = json.loads(resp_data)  # try to parse json results
+    except ValueError:
+        pass  # ignore json parsing errors
+    return resp_data, code, headers
```

Comment:
Here we need to return the `Replay-Nonce` so it can be used by the subsequent operation in the same thread.